src/Transport/Topology/Topologies/StandardTopology.cs

```diff
@@ -90,8 +90,11 @@ public Func<ICreateQueues> GetQueueCreatorFactory()
 
         public Func<CriticalError, IPushMessages> GetMessagePumpFactory()
         {
-            // todo, get criticial error integrated
-            return error => container.Resolve<IPushMessages>();
+            return error =>
+            {
+                container.Register<CriticalError>(() => error);
```

Comment:
@danielmarbach is there a way to get CriticalError as part of the `IPushMessages.Init` call from the core?